src/index.ts

```diff
@@ -194,8 +198,9 @@ const typescript: PluginImpl<RPT2Options> = (options) =>
 			return null;
 		},
 
-		transform(code, id)
+		async transform(code, id)
 		{
+			transformedFiles.add(id);
```

Comment:
I was thinking that this (or the `has` check below) might need normalization, but I actually don't think that's needed, because we use Rollup's native `id`s in both usages, so it's all Rollup and there's no TS <-> Rollup conversion needed.

A test with a circular import would be good to confirm that though.

I should probably add a comment here with that for future readers as well, since normalization is such a common gotcha we have deal with

Reply:
added an in-line comment